packages/editor-ui/src/modules/credentials.ts

```diff
@@ -0,0 +1,181 @@
+					}
+
+					return false;
+				});
+			};
+		},
+	},
+	actions: {
+		fetchCredentialTypes: async (context: ActionContext<ICredentialsState, IRootState>) => {
+			const credentialTypes = await getCredentialTypes(context.rootGetters.getRestApiContext);
+			context.commit('setCredentialTypes', credentialTypes);
+		},
+		fetchAllCredentials: async (context: ActionContext<ICredentialsState, IRootState>) => {
+			const credentials = await getAllCredentials(context.rootGetters.getRestApiContext);
+			context.commit('setCredentials', credentials);
+		},
+		getCredentialData: async (context: ActionContext<ICredentialsState, IRootState>, { id }: {id: string}) => {
+			return await getCredentialData(context.rootGetters.getRestApiContext, id);
+		},
+		createNewCredential: async (context: ActionContext<ICredentialsState, IRootState>, data: ICredentialsDecrypted) => {
+			const credential = await createNewCredential(context.rootGetters.getRestApiContext, data);
+			context.commit('upsertCredential', credential);
+
+			return credential;
+		},
+		updateCredential: async (context: ActionContext<ICredentialsState, IRootState>, params: {data: ICredentialsDecrypted, id: string}) => {
+			const { id, data } = params;
+			const credential = await updateCredential(context.rootGetters.getRestApiContext, id, data);
+			context.commit('upsertCredential', credential);
+
+			return credential;
+		},
+		deleteCredential: async (context: ActionContext<ICredentialsState, IRootState>, { id }: {id: string}) => {
+			const deleted = await deleteCredential(context.rootGetters.getRestApiContext, id);
+			if (deleted) {
+				context.commit('deleteCredential', id);
+			}
+		},
+		oAuth2Authorize: async (context: ActionContext<ICredentialsState, IRootState>, data: ICredentialsResponse) => {
+			return oAuth2CredentialAuthorize(context.rootGetters.getRestApiContext, data);
+		},
+		oAuth1Authorize: async (context: ActionContext<ICredentialsState, IRootState>, data: ICredentialsResponse) => {
+			return oAuth1CredentialAuthorize(context.rootGetters.getRestApiContext, data);
+		},
+		testCredential: async (context: ActionContext<ICredentialsState, IRootState>, data: ICredentialsDecrypted): Promise<NodeCredentialTestResult> => {
+			return testCredential(context.rootGetters.getRestApiContext, { credentials: data });
+		},
+		getNewCredentialName: async (context: ActionContext<ICredentialsState, IRootState>, params: { credentialTypeName: string }) => {
+			const { credentialTypeName } = params;
+
+			let newName = DEFAULT_CREDENTIAL_NAME;
+			if (!TYPES_WITH_DEFAULT_NAME.includes(credentialTypeName)) {
+				const { displayName } = context.getters.getCredentialTypeByName(credentialTypeName);
+				newName = getAppNameFromCredType(displayName);
+				newName = newName.length > 0 ? `${newName} ${DEFAULT_CREDENTIAL_POSTFIX}` : DEFAULT_CREDENTIAL_NAME;
+			}
+
+			try {
+				const res = await getCredentialsNewName(context.rootGetters.getRestApiContext, newName);
+				newName = res.name;
```

Comment:
Another option:
```suggestion
				return await getCredentialsNewName(context.rootGetters.getRestApiContext, newName).then(res => res.name);
```

Reply:
I prefer not to use `.then` for clarity but I will refactor this a bit.